src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -440,10 +438,10 @@ public static unsafe void AddScalarU(float scalar, Span<float> dst)
                     dstVector = Sse.Add(dstVector, scalarVector128);
                     Sse.Store(pDstCurrent, dstVector);
 
-                    pDstCurrent += SseIntrinsics.Vector128SingleElementCount;
+                    pDstCurrent += 4;
                 }
 
-                for (int i = 0; i < remainder - 4; i++)
+                for (int i = 0; i < remainder % 4; i++)
```

Comment:
I wouldn't use modulo operator here.

Maybe, do `remainder -= 4` in the above `if (remainder >= 4)` block and then your for loop can just be:
```C#
for (int i = 0; i < remainder; i++)
{
    ...
}
```